lib/ex_phone_number/metadata/phone_metadata.ex

```diff
@@ -43,41 +42,74 @@ defmodule ExPhoneNumber.Metadata.PhoneMetadata do
+           preferred_extn_prefix: ~x"./@preferredExtnPrefix"o |> transform_by(&normalize_string/1),
+           main_country_for_code: ~x"./@mainCountryForCode"o |> transform_by(&normalize_boolean/1),
+           mobile_number_portable_region: ~x"./@mobileNumberPortableRegion"o |> transform_by(&normalize_boolean/1),
+           carrier_code_formatting_rule: ~x"./@carrierCodeFormattingRule"s,
+           general: ~x"./generalDesc"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           fixed_line: ~x"./fixedLine"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           mobile: ~x"./mobile"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           toll_free: ~x"./tollFree"o |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           premium_rate: ~x"./premiumRate"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           shared_cost: ~x"./sharedCost"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           personal_number: ~x"./personalNumber"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           voip: ~x"./voip"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           pager: ~x"./pager"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           uan: ~x"./uan"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           voicemail: ~x"./voicemail"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           no_international_dialing: ~x"./noInternationalDialling"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           available_formats: [
+             ~x"./availableFormats/numberFormat"el,
+             number_format: ~x"."e |> transform_by(&NumberFormat.from_xpath_node/1)
+           ]
+         )
+
+    kwlist = if(is_map(kwlist) && is_map(kwlist.general)) do
+      set_general_possible_lengths(kwlist)
+    else
+      kwlist
+    end
+
     struct(%PhoneMetadata{}, kwlist)
   end
 
+  defp set_general_possible_lengths(kwlist) do
+    possible_lengths =
+      [
+        Map.get(kwlist, :fixed_line) |> get_possible_length,
+        Map.get(kwlist, :mobile) |> get_possible_length,
+        Map.get(kwlist, :toll_free) |> get_possible_length,
+        Map.get(kwlist, :premium_rate) |> get_possible_length,
+        Map.get(kwlist, :shared_cost) |> get_possible_length,
+        Map.get(kwlist, :personal_number) |> get_possible_length,
+        Map.get(kwlist, :voip) |> get_possible_length,
+        Map.get(kwlist, :pager) |> get_possible_length,
+        Map.get(kwlist, :uan) |> get_possible_length,
+        Map.get(kwlist, :voicemail) |> get_possible_length,
+        Map.get(kwlist, :no_international_dialing) |> get_possible_length,
+      ]
+      |> Enum.reject(&is_nil/1)
+      |> List.flatten
+      |> Enum.sort
+      |> Enum.dedup
+
+    put_in(kwlist.general.possible_lengths, possible_lengths)
+  end
+
+  defp get_possible_length(nil), do: nil
```

Comment:
If `defp get_possible_length(nil), do: []` (returning empty array instead of `nil`), then it makes above code easier to manage. Similar can be applied for `phone_number_description.possible_lengths || []`.